python-avd/pyavd/_eos_designs/structured_config/base/platform_mixin.py

```diff
@@ -0,0 +1,132 @@
+                if intf_profile_settings is not None:
+                    sfe_profile_member_interfaces.append(intf_profile_settings)
+
+        if sfe_profile_member_interfaces:
+            return sfe_profile_member_interfaces._natural_sorted()
+        return None
+
+    def _build_interface_with_rx_queue_settings(
+        self: AvdStructuredConfigBaseProtocol,
+        l3_member_interface: EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3InterfacesItem
+        | EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3PortChannelsItem.MemberInterfacesItem,
+    ) -> EosCliConfigGen.Platform.Sfe.Interface.ProfilesItem.InterfacesItem | None:
+        """Returns one SFE interface with profile settings for structured config."""
+        if not l3_member_interface._get("rx_queue"):
+            # specified interface does not have explicit "rx_queue" key or sub-keys specified
+            # exclude such interface from profile being built
+            return None
+        # validate rx_queue 'count' when specified
+        if l3_member_interface.rx_queue._get("count") and l3_member_interface.rx_queue.count > self.shared_utils.max_rx_queues:
+            msg = (
+                f"'rx_queue' count for interface '{l3_member_interface.name}' exceeds maximum supported '{self.shared_utils.max_rx_queues}' for this platform."
+            )
+            raise AristaAvdInvalidInputsError(msg)
+
+        rx_queue_workers = set()
+        for _worker in l3_member_interface.rx_queue.worker:
+            for worker_id in range_expand(_worker):
+                self._validate_rx_queue_worker(worker_id, _worker, l3_member_interface.name)
+                rx_queue_workers.add(int(worker_id))
+        rx_queue_workers_range = list_compress(list(rx_queue_workers))
+
+        intf_profile_settings = EosCliConfigGen.Platform.Sfe.Interface.ProfilesItem.InterfacesItem()
+        intf_profile_settings.name = l3_member_interface.name
+        intf_profile_settings.rx_queue.count = l3_member_interface.rx_queue.count
+        if rx_queue_workers_range:
+            intf_profile_settings.rx_queue.worker = rx_queue_workers_range
+        if l3_member_interface.rx_queue._get("mode"):
+            # "mode" has default value, include value only when explicitly specified
+            intf_profile_settings.rx_queue.mode = l3_member_interface.rx_queue.mode
+        return intf_profile_settings
+
+    def _validate_rx_queue_worker(self: AvdStructuredConfigBaseProtocol, worker_id: str, worker_range: str, l3_member_interface_name: str) -> None:
+        if int(worker_id) >= self.shared_utils.max_rx_queues:
+            msg = (
+                f"One or more worker ids within '{worker_range}' under 'rx_queue' for interface '{l3_member_interface_name}' "
+                f"equal or exceed maximum supported '{self.shared_utils.max_rx_queues}' for this platform."
+            )
+            raise AristaAvdInvalidInputsError(msg)
```

Comment:
I think it is enough to just tell which worker number was above the limit and not care about how it was given.